src/Distributor/MostExpensiveFirstDistributor.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusPromotionExtensionsPlugin\Distributor;
+
+use Webmozart\Assert\Assert;
+
+final class MostExpensiveFirstDistributor implements MostExpensiveFirstDistributorInterface
+{
+    public function distribute(array $integers, int $amount): array
+    {
+        Assert::allInteger($integers);
+
+        // Do a first round with only first item to eliminate most of non-applicable items
+        $firstItemWithPromotion = ($integers[0] - $amount);
```

Comment:
I guess there is some presumed order of these integers. This isn't enforced in this piece of code which is both a source of misunderstanding, but also a bug further down the line. Unless I am mistaken?